tests/components/transmission/test_init.py

```diff
@@ -184,3 +169,27 @@ async def test_migrate_unique_id(
 
     assert migrated_entity
     assert migrated_entity.unique_id == new_unique_id
+
+
+async def test_coordinator_update_error(
+    hass: HomeAssistant,
+    mock_transmission_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test coordinator handles TransmissionError during update."""
+    client = mock_transmission_client.return_value
+
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    coordinator = mock_config_entry.runtime_data
+    assert coordinator.last_update_success
+
+    client.session_stats.side_effect = TransmissionError("Connection failed")
+
+    await coordinator.async_refresh()
```

Comment:
We shouldn't touch internals. Instead we should test what the user sees about this event. As in, if we can't fetch data, we should render entities unavailable, and we should test that behaviour instead